### PR TITLE
executor: migrate test infra to testify for inspection_summary_test.go

### LINKED ISSUE #28591: migrate test-infra to testify for `inspection_summary_test.go`

### PR DESCRIPTION
**Pull request by YufeiCh**
### What problem does this PR solve?

Issue Number: close #28591 

Problem Summary:

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: executor/inspection_summary_test.go [hunk 1]
```diff
@@ -16,62 +16,45 @@ package executor_test
...
+func TestInspectionSummary(t *testing.T) {
+	store, domain, clean := testkit.CreateMockStoreAndDomain(t)
+	defer clean()
+
+	tk := testkit.NewTestKit(t, store)
 
 	fpName := "github.com/pingcap/tidb/executor/mockMetricsTableData"
-	c.Assert(failpoint.Enable(fpName, "return"), IsNil)
-	defer func() { c.Assert(failpoint.Disable(fpName), IsNil) }()
+	require.Nil(t, failpoint.Enable(fpName, "return"))
```
Comment (hawkingrei) [inline executor/inspection_summary_test.go:51]: use ```require.NoError```

**Review by hawkingrei — APPROVED** [review]

**Comment by hawkingrei** [comment]
Please fix the linter.

**Review by tangenta — APPROVED** [review]

**Pull request merged**